html/modules/custom/bc_dc/src/Controller/BcDcCreateFileController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Drupal\bc_dc\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use vendor\PhpOffice\PhpSpreadsheet\Writer;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use vendor\PhpOffice\PhpSpreadsheet\Reader\Xlsx;
+
+class BcDcCreateFileController extends ControllerBase {
+
+  /**
+   * Returns a downloadable file.
+   */
+  public function createFile($node, $param) {
+  
+    if ($param === 'csv' || $param === 'xlsx') {
```

Comment:
Have an array of allowed types. If `$param` is not in the array, throw the exception. It is difficult to understand an `if` statement that is so long. It is easier to read to have short conditions near the start that return to handle the error states and then have the body of the function not inside any `if` statements.

Reply:
fixed